OpenSim/Common/Component.cpp

```diff
@@ -1035,6 +1033,25 @@ void Component::setNextSubcomponentInSystem(const Component& sub) const
     }
 }
 
+void Component::updateFromXMLNode(SimTK::Xml::Element& node, int versionNumber)
+{
+    if (versionNumber < XMLDocument::getLatestVersion()) {
+        if (versionNumber < 30508) {
+            // Here's an example of the change this function might make:
+            // Previous: <connectors>
+            //               <Connector_PhysicalFrame_ name="parent">
+            //                   <connectee_name>...</connectee_name>
+            //               </Connector_PhysicalFrame_>
+            //           </connectors>
+            // New:      <connector_parent_connectee_name>...
+            //               </connector_parent_connectee_name>
+            //
```

Comment:
> Previously, the Component had a property list of Connectors, which kept Connectors grouped.

We may be able to edit the serialization code to first serialize properties whose names contain `connectee_name`. What do you think?

I agree that this is definitely a downside, but if we must live with it: I feel that the proposed design has fewer downsides than the alternatives.